samples/nrf9160/nrf_cloud_mqtt_multi_service/src/application.c

```diff
@@ -176,8 +181,93 @@ static void on_location_update(const struct location_event_data * const location
 	}
 }
 
+/**
+ * @brief Receives general device messages from nRF Cloud, checks if they are AT command requests,
+ * and performs them if so, transmitting the modem response back to nRF Cloud.
+ *
+ * Try sending {"appId":"MODEM", "messageType":"CMD", "data":"AT+CGMR"}
+ * in the nRF Cloud Portal Terminal card.
+ *
+ * @param msg - The device message to check.
+ */
+static char at_req_resp_buf[CONFIG_AT_HOST_CMD_MAX_LEN];
+static void handle_at_cmd_requests(const char * const msg) {
+
+	/* Attempt to parse the message as if it is JSON */
+	struct cJSON *msg_obj = cJSON_Parse(msg);
+
+	if (!msg_obj) {
+		/* The message isn't JSON or otherwise couldn't be parsed. */
+		return;
+	}
+
+	/* Check whether the JSON device message is a modem AT command */
+	if (strcmp(cJSON_GetStringValue(cJSON_GetObjectItem(msg_obj, NRF_CLOUD_JSON_APPID_KEY)),
+		   NRF_CLOUD_JSON_APPID_VAL_MODEM) != 0) {
+		goto cleanup;
+	}
+	if (strcmp(cJSON_GetStringValue(cJSON_GetObjectItem(msg_obj, NRF_CLOUD_JSON_MSG_TYPE_KEY)),
+		   NRF_CLOUD_JSON_MSG_TYPE_VAL_CMD) != 0) {
+		goto cleanup;
+	}
+
+	/* If it is, extract the requested command string */
+	const char *cmd = cJSON_GetStringValue(cJSON_GetObjectItem(msg_obj, "data"));
```

Comment:
you could use `NRF_CLOUD_JSON_DATA_KEY`.
but i guess this specific string is user generated, not cloud generated... 🤷

Reply:
No, you're right -- I missed this line when replacing my handwritten consts during cleanup, good ctach